src/middleware/navigation/check.condition.ts

```diff
@@ -72,6 +72,10 @@ export const checkUpdatePresenterEntered = (appData: ApplicationData): boolean =
   return checkOverseasEntityNumberEntered(appData) && Object.keys(appData[PresenterKey] || {}).length !== 0;
 };
 
+export const checkEntityUpdateDetailsEntered = (appData: ApplicationData): boolean => {
+  return checkUpdatePresenterEntered(appData) && Object.keys(appData[EntityKey] || {}).length !== 0;
```

Comment:
This seems not right. As checking presenter when supposed to be about entity? Is this merged with Cory's fix?

Reply:
The Register journey implements this recursive kind of check to ensure you don't land on the page without selecting any previous values.

Reply:
maybe it's checkEntityAndPresenterDeatiledEntered then?

Reply:
Renaming up to you